core/src/main/resources/persistence.proto

```diff
@@ -0,0 +1,71 @@
+package persistence;
+
+message UserObject {
+    required bytes wrappedObject = 1;
+}
+
+message WrappedByteArray {
+    required bytes wrappedBytes = 1;
+    optional int32 hashcode = 2;
+}
+
+message KeyValuePair {
+    optional bytes key = 1;
```

Comment:
are key and value really optional ?

Reply:
There are many cases in the testsuite where a null key/value is used, so I went with this. But thinking about it the fields should be required here and a an empty `byte[]` passed in the aforementioned cases.

Reply:
well, if these are truly nullable, then let's keep them optional without empty byte[] tricks